Services/Certificate/classes/class.ilCertificateMigrationGUI.php

```diff
@@ -144,7 +144,7 @@ public function startMigrationAndReturnMessage(): string
 			$this->user, new \ilCertificateMigration($this->user->getId())
 		);
 		if (false === $isMigrationAvailable) {
-			$this->errorHandler->raiseError($this->lng->txt('permission_denied'), $this->errorHandler->MESSAGE);
+			return '';
```

Comment:
Maybe a ...

```php
\ilUtil::sendFailure($this->lng->txt('permission_denied'), true);
$this->ctrl->redirectByClass(['ilPersonalDesktopGUI', 'ilPersonalProfileGUI']);
```

... also works?

Reply:
I think I would prefer a combination: The message that @legionth wrote to the logger, also as failure message and the redirect to the personalprofilegui. What do you think?

Reply:
I talked with @mjansenDatabay about this. The problem is that the Background Tasks "delete" action has the last performed location in its link. Which means the command `startMigrationAndReturnMessage` will be executed when deleting the background task, right after the migration.

This behavior appears only right after the migration. If the site changed (e.g. open a form or some view) and this problem disappears.
I talked with @mjansenDatabay about this and we think we shouldn't throw an visible error here, because it's not really an error and this can only occure once per user and only if the user doesn't change the site.